### PR TITLE
clamav: upgrade to 1.0.6

### PR DESCRIPTION
**Pull request by BettyRain**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
clamav: upgrade to 1.0.6

###### Change Log  

- Upgrade clamav package to 1.0.6

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Local build
- Pipeline build id: [557120](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=557120&view=results)

### CODE REVIEW

File: SPECS/clamav/clamav.spec [hunk 1]
```diff
@@ -137,8 +139,8 @@ fi
 %dir %attr(-,clamav,clamav) %{_sharedstatedir}/clamav
 
 %changelog
-* Wed Apr 17 2024 Andrew Phelps <anphel@microsoft.com> - 0.105.2-5
-- Fix build break by adding BR for systemd
+* Thu Apr 18 2024 Betty Lakes <bettylakes@microsoft.com> - 1.3.1-1
```
Comment (anphel31) [inline SPECS/clamav/clamav.spec:142]: looks like part of the changelog was accidentally overwritten?

File: SPECS/clamav/clamav.spec [hunk 2]
```diff
@@ -126,6 +127,7 @@ fi
 %{_libdir}/*.so.*
 %{_libdir}/pkgconfig/*.pc
 %{_libdir}/systemd/system/*
+%{_libdir}/*.a
```
Comment (anphel31) [inline SPECS/clamav/clamav.spec:130]: This is likely not needed and can be excluded (especially in the core package)

File: SPECS/clamav/clamav.spec [hunk 3]
```diff
@@ -77,11 +77,12 @@ cmake \
     -D ENABLE_SYSTEMD=ON \
     -D ENABLE_MILTER=OFF \
     -D ENABLE_EXAMPLES=OFF
-%cmake_build
+%cmake3_build
 
 %check
-cd build
-ctest --verbose
+%ctest3 -- -E valgrind
```
Comment (anphel31) [inline SPECS/clamav/clamav.spec:83]: why run the tests twice?

File: SPECS/clamav/clamav.spec (hunk 1 above)
Comment (BettyRain) [reply SPECS/clamav/clamav.spec:142]: Oh, right, weird

File: SPECS/clamav/clamav.spec (hunk 2 above)
Comment (BettyRain) [reply SPECS/clamav/clamav.spec:130]: There is a reason why I added this file:
image
So, would you like me to add the file in a less wide way?
%{_libdir}/*.a -> %{_libdir}/*_rust.a or smth like %{_libdir}/lib%{name}*.a
Or make this file be removed somewhere earlier?

**Comment by eric-desrochers** [comment]
Let's stick to LTS version `1.0`

https://endoflife.date/clamav
image

File: SPECS/clamav/clamav.spec (hunk 2 above)
Comment (BettyRain) [reply SPECS/clamav/clamav.spec]: I currently changed it to  %{_libdir}/lib%{name}*.a, let me know what you think

Comment (BettyRain) [reply SPECS/clamav/clamav.spec]: Removed it after upgrading to 1.0.6

**Review by anphel31 — APPROVED** [review]

File: SPECS/clamav/clamav.spec (hunk 2 above)
Comment (anphel31) [reply SPECS/clamav/clamav.spec]: I was suggesting to simply delete the file and not package it. But it looks like this is not necessary for the 1.0.6 version

**Pull request merged**